api/v1alpha1/smbshare_types.go

```diff
@@ -117,6 +117,10 @@ type SmbShareStatus struct {
 // +kubebuilder:object:root=true
 // +kubebuilder:subresource:status
 
+// +kubebuilder:printcolumn:JSONPath=`.spec.shareName`,description="Name of the Samba share",name="Share-name",type=string
+// +kubebuilder:printcolumn:JSONPath=`.spec.storage.pvc.path`,description="Path for the share within PVC",name="Share-path",type=string
+// +kubebuilder:printcolumn:JSONPath=`.spec.scaling.availabilityMode`,description="Samba availability mode",name="Availability",type=string
+// +kubebuilder:printcolumn:JSONPath=`.metadata.creationTimestamp`,name="Age",type=date
```

Comment:
The linter is conflicting with the kubebuilder comments. You can try to change the order of things and see if that makes the linter and kubebuilder both happy. Otherwise, you can add a special comment to disable the linter(s) for this block of code.

Otherwise, mostly LGTM and we ought to be able to merge it after we resolve this linting issue.

Reply:
An empty new line could resolve linter error :-)

For character limit errors from:
* `revive` - disabled the specific rule for the block using _revive:disable:line-length-limit_ comment
* `golangci-lint` - disabled using _nolint:lll_ comment